designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/validation/PrettyValidationErrors.scala

```diff
@@ -42,27 +37,82 @@ object PrettyValidationErrors {
-          "Nodes cannot have empty id",
-          errorType = NodeValidationErrorType.RenderNotAllowed
-        )
+      case propertiesError: ScenarioPropertiesError =>
+        propertiesError match {
+          case EmptyScenarioId(isFragment) =>
+            node(
+              s"${if (isFragment) "Fragment" else "Scenario"} name is mandatory and cannot be empty",
+              s"Empty ${if (isFragment) "fragment" else "scenario"}",
+              fieldName = Some("$id")
+            )
+          case BlankScenarioId(isFragment) =>
+            node(
+              s"${if (isFragment) "Fragment" else "Scenario"} name cannot be blank",
+              s"Blank ${if (isFragment) "fragment" else "scenario"} name",
+              fieldName = Some("$id")
+            )
+          case LeadingSpacesScenarioId(isFragment) =>
+            node(
+              s"${if (isFragment) "Fragment" else "Scenario"} name cannot have leading spaces",
+              s"Leading spaces in ${if (isFragment) "fragment" else "scenario"} name",
+              fieldName = Some("$id")
+            )
+          case TrailingSpacesScenarioId(isFragment) =>
+            node(
+              s"${if (isFragment) "Fragment" else "Scenario"} name cannot have trailing spaces",
+              s"Trailing spaces in ${if (isFragment) "fragment" else "scenario"} name",
+              fieldName = Some("$id")
+            )
+          case ScenarioNameValidationError(message, description) =>
+            node(
+              message,
+              description,
+              fieldName = Some("$id"),
+            )
+          case SpecificDataValidationError(field, message) => node(message, message, fieldName = Some(field))
+        }
+      case nodeIdError: NodeIdError =>
+        nodeIdError match {
+          case EmptyNodeId() =>
+            node(
+              "Node name cannot be empty",
+              "Empty node name",
+              errorType = NodeValidationErrorType.RenderNotAllowed,
+              fieldName = Some("$id")
+            )
+          case InvalidCharactersNodeId(_) =>
+            node(
+              "Node name contains invalid characters. Quotation mark (\"), comma (,) and apostrophe (') are not allowed",
+              "Invalid characters in node name",
+              NodeValidationErrorType.RenderNotAllowed,
+              Some("$id")
+            )
+          case BlankNodeId(_) =>
+            node(
+              "Node name cannot be blank",
+              "Blank node name",
+              fieldName = Some("$id")
+            )
+          case LeadingSpacesNodeId(_) =>
+            node(
+              "Node name cannot have leading spaces",
+              "Leading spaces in node name",
+              fieldName = Some("$id")
+            )
+          case TrailingSpacesNodeId(_) =>
+            node(
+              "Node name cannot have trailing spaces",
+              "Trailing spaces in node name",
+              fieldName = Some("$id")
```

Comment:
Let's keep these special field names in some one place. We already do a similar thing with expressions in `NodeExpressionId`

Reply:
Good point, I think the best place for it would be in `scenario-api` in `CanonicalProcess` and `node` accordingly, added it there.